### PR TITLE
Buffer change and ignoring unknown entries

### PR DESCRIPTION
**Pull request by moimart**
Hi,

I'm putting both changes together because I can't find an icon that does not have any of the conflicting entries; hence, a bit unsure when testing. I tested both commits and it works.

So now it's more cleanly ignoring entries based on what's not in Icns.TYPES; no more blacklist :)

Cheers,
Moises

### CODE REVIEW

File: lib/entry.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ function IconEntry() {
   this.isCompressed = false
   this.description = null
   this.osVersion = null
+  this.validEntry = true;
```
[the remark this answers is not in the document]
Comment (moimart) [reply lib/entry.js]: yes. it aligns better with the isCompressed and the rest

**Comment by moimart** [comment]
Changes in new commit

**Comment by jhermsmeier** [comment]
Awesome, thanks again!

**Comment by jhermsmeier** [comment]
Published as `apple-icns@0.2.0`

**Comment by moimart** [comment]
Thanks a lot!

**Pull request merged**